plugins/web-worker-offloading/third-party/google-site-kit.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Web Worker Offloading integration with Site Kit by Google.
+ *
+ * @since n.e.x.t
+ * @package web-worker-offloading
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Configures WWO for Site Kit and Google Analytics.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @link https://partytown.builder.io/google-tag-manager#forward-events
+ *
+ * @param array<string, mixed>|mixed $configuration Configuration.
+ * @return array<string, mixed> Configuration.
+ */
+function plwwo_google_site_kit_configure( $configuration ): array {
+	$configuration = (array) $configuration;
+
+	$configuration['globalFns'][] = 'gtag'; // Because gtag() is defined in one script and called in another.
+	$configuration['forward'][]   = 'dataLayer.push'; // Because the Partytown integration has this in its example config.
+
+	// See <https://github.com/google/site-kit-wp/blob/440dcd6e8289a8776acc60ecb3c13f99bfd0ac5a/includes/Core/Consent_Mode/Consent_Mode.php#L239-L240>.
+	$configuration['mainWindowAccessors'][] = '_googlesitekitConsentCategoryMap';
+	$configuration['mainWindowAccessors'][] = '_googlesitekitConsents';
+
+	return $configuration;
+}
+add_filter( 'plwwo_configuration', 'plwwo_google_site_kit_configure' );
+
+plwwo_mark_scripts_for_offloading(
+	array(
+		'google_gtagjs',
+		'googlesitekit-consent-mode',
```

Comment:
I don't know how the Consent Mode is supposed to work, especially as it relates to the WP Consent API plugin. On my test site I didn't get any consent banner even without WWO active, so I don't know what to look for to see if it is working as expected.

Reply:
you may need to have this plugin installed? https://wordpress.org/plugins/wp-consent-api/

Reply:
cc: @aaemnnosttv ^^